src/main/java/ch/njol/skript/expressions/ExprFurnaceTime.java

```diff
@@ -0,0 +1,297 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Getter;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Block;
+import org.bukkit.block.Furnace;
+import org.bukkit.event.Event;
+import org.bukkit.event.inventory.FurnaceBurnEvent;
+import org.bukkit.event.inventory.FurnaceExtractEvent;
+import org.bukkit.event.inventory.FurnaceSmeltEvent;
+import org.bukkit.event.inventory.FurnaceStartSmeltEvent;
+import org.jetbrains.annotations.Nullable;
+
+import static ch.njol.util.Math2.floor;
+
+@Name("Furnace Times")
+@Description({
+	"The cook time, total cook time, and burn time of a furnace. Can be changed.",
+	"<ul>",
+	"<li>cook time: The amount of time an item has been smelting for.</li>",
+	"<li>total cook time: The amount of time an item is required to smelt for until done.</li>",
+	"<li>burn time: The amount of time left of the fuel.</li>",
```

Comment:
```suggestion
	"<li>burn time: The amount of time left for the current fuel.</li>",
```
you should clarify if this is for the whole stack or just the current item